lightning/src/ln/channel.rs

```diff
@@ -2969,6 +2969,27 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before to propose a feerate update, check that we can actually afford the new fee compare to the holder balance.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (inbound_stats.pending_htlcs as u64 + outbound_stats.pending_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
+		if self.value_to_self_msat / 1000 - total_fee - outbound_stats.pending_htlcs_value_msat < self.counterparty_selected_channel_reserve_satoshis.unwrap() {
```

Comment:
I think `outbound_stats.pending_htlcs_value_msat` also needs to be divided by 1000?

Reply:
Oh no that sounds right to me...Corrected, thanks for catching that!